prov/verbs/src/verbs_mr.c

```diff
@@ -73,7 +98,14 @@ int vrb_mr_reg_common(struct vrb_mem_desc *md, int vrb_access, const void *buf,
 	if (md->domain->flags & VRB_USE_ODP && iface == FI_HMEM_SYSTEM)
 		vrb_access |= VRB_ACCESS_ON_DEMAND;
 
-	md->mr = ibv_reg_mr(md->domain->pd, (void *) buf, len, vrb_access);
+#if VERBS_HAVE_DMABUF_MR
+	if (iface == FI_HMEM_ZE && ze_is_addr_valid(buf))
```

Comment:
This is unrelated to this PR, but what does the ze_is_addr_valid() check?  Because the address is obviously valid.

Reply:
It checks if the address is device memory. Since host memory can also be allocated via oneAPI L0, passing such address down here should be allowed. The check is optional, though. Because host memory allocated in this way should also work with ibv_reg_dmabuf_mr().

Reply:
Oh, wait, this is a direct call to ze, and not our abstraction.  It's their function naming that is confusing, not ours.  :)

Reply:
No, it's our abstraction, defined inside src/hmem_ze.c. We can change the name :)

Reply:
In that case, ze_is_device_addr() seems like an improvement.

Reply:
Wait, it's used in filling out "ofi_hmem_ops", where the member function is named "is_addr_valid".

Reply:
I think we can just drop the call here.